### ISSUE TITLE
Capture exit codes for pipes, not just the last command of a pipe.

### ISSUE THREAD

**Issue by docwhat** [bhilburn/powerlevel9k#514]
The current implementation only shows the exit code for the last command in a command with pipes.

Instead, I would suggest showing all the exit codes.

Current behavior (assuming `PIPE_FAIL` isn't set):

``` sh
$ true | false | false | true # Shows checkmark
$ true | false | false # Shows 1 ↵
```

Instead, the `status` segment should show:

```
$ true | false | false | true # Shows 0|1|1|0 ↵
$ true | false | false # Shows 0|1|1 ↵
```

Potentially, `0` could be a checkmark.

To do this, you need to capture `$pipestatus[@]`.

**Comment by bhilburn** [comment]
@docwhat - This is a fantastic idea! Thanks so much for sharing the idea & creating a PR! Moving the discussion to #515.

**Comment by bhilburn** [comment]
Closed by merge of #515!
